planetarypraxis/smartforests
# Geographic map of pages

## What it does

- Adds React / typescript build configuration with an entrypoint in `smartforests/typescript/index.ts`
- Adds `MapPage` to the Django app
- Adds accompanying React / Mapbox app which displays Wagtail pages that have geographic coordinates
- You can hover over and click into these pages and a sidebar appears where more data can be displayed

<img width="1441" alt="Screenshot 2021-10-15 at 15 34 58" src="https://user-images.githubusercontent.com/237556/137504814-647306c4-2d2a-4aab-a939-c8a52f417dad.png">

## How to test this PR

1. Copy `.env.template` to `.env` at the root and add the [Mapbox developer public key](https://account.mapbox.com/access-tokens/ckur0w84z1nj52vp1jjjwztc3).
1. Migrate the Django app
2. Ensure Webpack has built the assets (e.g. through watch mode)
3. Add a `MapPage` in the CMS admin panel
4. Navigate to the page and you'll see a map!
5. Bonus: can you test production deployments locally?
